usl_pipeline/usl_lib/usl_lib/storage/metastore.py

```diff
@@ -117,6 +120,41 @@ def update_min_max_elevation(
         transaction = db.transaction()
         _update_study_area_min_max_elevation(transaction, study_area_ref, min_, max_)
 
+    @staticmethod
+    def update_chunk_info(
+        db: firestore.Client, study_area_name: str, chunk_size: int
+    ) -> None:
+        """Sets chunk-related properties of study area if they weren't set before.
+
+        Args:
+          db: The firestore database client to use to make the update.
+          study_area_name: The study area to update.
+          chunk_size: Size of the chunk in cells. Will only be set chunk-related fields
+                if they weren't set before.
+        """
+        study_area = StudyArea.get(db, study_area_name)
+        if study_area.chunk_size is not None:
+            if study_area.chunk_size != chunk_size:
+                raise ValueError(
+                    "Unexpected variation detected for chunk size:"
+                    + f" {study_area.chunk_size} != {chunk_size}"
+                )
+            return
+
+        # Calculate number of chunks over each of X and Y axes:
+        x_count = (study_area.col_count + chunk_size - 1) // chunk_size
+        y_count = (study_area.row_count + chunk_size - 1) // chunk_size
+
+        study_area_ref = db.collection(STUDY_AREAS).document(study_area_name)
+        db.transaction().update(
+            study_area_ref,
+            {
+                "chunk_size": chunk_size,
+                "chunk_x_count": x_count,
+                "chunk_y_count": y_count,
+            },
```

Comment:
`ref.set({...}, merge=True)`
will give up update semantics (only setting these three fields, but leaving any other fields alone) without a transaction.

Reply:
I decided to use `ref.update` instead since `set` gives you an advantage of creating the doc if it didn't exist before (which I don't need).